src/states/SelectAttackState.py

```diff
@@ -103,34 +109,37 @@ def update(self, dt, events):
                         if self.selectAttackTile > len(self.avilableAttackTile) - 1:
                             self.selectAttackTile = 0
                 if event.key == pygame.K_RETURN:
-                    if self.effectOwner == PlayerType.PLAYER:
-                        if self.selectAttackTile>=0 and self.effect.maxRange>0:
-                            # ATTACK
-                            if self.field[self.avilableAttackTile[self.selectAttackTile]].is_occupied():
+                    if self.selectAttackTile>=0 and self.effect.maxRange>0:
+                        # ATTACK
+                        if self.field[self.avilableAttackTile[self.selectAttackTile]].is_occupied():
+                            if self.effectOwner == PlayerType.PLAYER:
                                 self.player.ChangeAnimation("multi_attack")
                                 self.enemy.ChangeAnimation("death")
-                                damage = self.player.attack - self.field[self.avilableAttackTile[self.selectAttackTile]].entity.defense
-                                if damage > 0:
-                                    # ATTACK HIT
-                                    gSounds['attack'].play()
-                                    self.field[self.avilableAttackTile[self.selectAttackTile]].entity.health -= damage
-                                    self.field[self.avilableAttackTile[self.selectAttackTile]].entity.stunt = True
-                                    print(f'{self.field[self.avilableAttackTile[self.selectAttackTile]].entity} takes {damage} damage')
-                                    if self.effect.type == EffectType.ATTACK_SELF_BUFF:
-                                        buffList = self.getBuffListFromEffect(self.effect)
-                                        self.player.add_buffs(buffList)
-                                    if self.effect.type == EffectType.ATTACK_OPPO_BUFF:
-                                        buffList = self.getBuffListFromEffect(self.effect)
-                                        self.enemy.add_buffs(buffList)
-                                else:
-                                    # ATTACK BLOCK
-                                    gSounds['block'].play()
-                                    print(f'{self.field[self.avilableAttackTile[self.selectAttackTile]].entity} takes no damage')
-                                self.field[self.avilableAttackTile[self.selectAttackTile]].entity.print_stats()
                             else:
-                                print("no entity on the targeted tile")
+                                self.enemy.ChangeAnimation("attack")
+                                self.player.ChangeAnimation("knockdown")
```

Comment:
should the animation be played only when the attack is hit? I think there are two ways to play the animation
- the attack animation is played immediately / the knockdown is played when attack is hit
- both attack and knockdown animation will be played only when attack is hit

Reply:
the attack animation is played immediately / the knockdown is played when attack is hit